packages/internal/identity-resolver/src/identity-resolver.ts

```diff
@@ -50,15 +48,38 @@ export class IdentityResolver {
     return {
       did: document.id,
       pds: new URL(service.serviceEndpoint),
-      handle: extractHandle(document),
+      handle: getValidHandle(document),
     }
   }
 
+  public async getDocument(
+    input: string,
+    options?: ResolveIdentityOptions,
+  ): Promise<DidDocument<AtprotoIdentityDidMethods>> {
+    return isResolvedHandle(input)
+      ? this.getDocumentFromDid(input, options)
+      : this.getDocumentFromHandle(input, options)
+  }
+
   public async getDocumentFromDid(
     did: NonNullable<ResolvedHandle>,
     options?: ResolveDidOptions,
   ): Promise<DidDocument<AtprotoIdentityDidMethods>> {
-    return this.didResolver.resolve(did, options)
+    const document = await this.didResolver.resolve(did, options)
+
+    const handle = getValidHandle(document)
+    if (handle) {
+      options?.signal?.throwIfAborted()
+
+      const did = await this.handleResolver.resolve(handle, options)
+      if (did !== document.id) {
+        throw new TypeError(
+          `Did document for "${did}" does not match the handle "${handle}"`,
+        )
+      }
+    }
+
+    return document
   }
```

Comment:
I don't think this is the behavior we want for atproto DID resolution.  The status of your handle doesn't make your identity any more or less legitimate, and we perform many operations on your identity without a valid handle (e.g. validating your records). One reason we use strong links based on DID is to avoid reliance on handle status.

A "resolved handle" is a DID, but a DID isn't a resolved handle.  So I wonder if the more appropriate input type here might be `AtprotoDid`.

Reply:
When signing in, the user can either provide a `handle`, `did` or `pdsOrigin`. In case they provide a `did`, this change will ensure that the did, if it has a handle, resolves bi-dirrectionaly.

This method can also be used as a reliable way to determine a `did -> handle` mapping